game/graphic/texture_loader.cc

```diff
@@ -0,0 +1,190 @@
+#include "texture_loader.hh"
+
+#include "glutil.hh"
+
+#include <atomic>
+#include <condition_variable>
+#include <functional>
+#include <map>
+#include <mutex>
+#include <thread>
+
+namespace {
+
+    class TextureLoader {
+    public:
+        static TextureLoader& instance() {
+            if (!m_instance)
+                m_instance = std::make_unique<TextureLoader>();
+
+            return *m_instance;
+        }
+        static void destroy() {
+            m_instance.reset();
+        }

```

Comment:
Hmmm singleton here, do we need it to be singleton? or is it possible to make it transient (everytime new when called and going out of scope when not needed anymore) lifetime so bitmaps get disposed when they're not in use anymore? this also prevents a migration to possible dependency injection

Reply:
The TextureLoader was a singleton before this. So I decided not to change it to much. And we could remove it, but that will be a whole bunch of changes. Then every place where we load/define a texture must be modified.

Reply:
And there are still too many places where the singleton is needed. Would be nice but it is not the scope of this pr